core/src/cubos/core/ecs/dispatcher.cpp

```diff
@@ -186,6 +186,15 @@ bool Dispatcher::dfsVisit(DFSNode& node, std::vector<DFSNode>& nodes)
 
 void Dispatcher::callSystems(World& world, CommandBuffer& cmds)
 {
+    // If the systems haven't been prepared yet, do so now.
+    // We can't multi-thread this as the systems require exclusive access to the world to prepare.
+    if (!this->prepared)
```

Comment:
Since this runs only one, we could do this in a startup system instead in the future.

Reply:
Hm, even startup systems may have state on their arguments, so maybe its better to keep the logic separated.
This way its also easy to enforce an order on the setup of that state.